codalab/model/bundle_model.py

```diff
@@ -829,13 +829,19 @@ def transition_bundle_starting(self, bundle, user_id, worker_id):
             Adds a worker_run row that tracks which worker will run the bundle.
         """
         with self.engine.begin() as connection:
-            # Check that it still exists.
+            # Check if the requested bundle still exists.
             row = connection.execute(
                 cl_bundle.select().where(cl_bundle.c.id == bundle.id)
             ).fetchone()
             if not row:
                 # The user deleted the bundle.
                 return False
+            # Check if the designated worker still exists
```

Comment:
Is this necessary / related to this PR?

Reply:
The intention of adding this logic here is to handle the case when there are a large number of staged jobs waiting to be dispatched, the real dispatching logic will take longer than usual as follows https://github.com/codalab/codalab-worksheets/blob/dad99c952ecdc541cdc8b9a6f594ac2741c3086b/codalab/server/bundle_manager.py#L384-L408
However, the `WorkerInfoAccessor` object is fetched at the beginning of each scheduling loop. If there is any worker terminated in the middle of the dispatching loop,  the `WorkerInfoAccessor` object won't get refreshed. Then the worker terminated earlier will still present in the available `workers_list` and bundle manager will still try to assign jobs to this worker even though it's already terminated.

Reply:
I removed this one just now, coz this case can be handled by restaging bundles in the `starting` logic after timeout period (5 minutes).